packages/salesforce-adapter/src/filters/remove_fields_and_values.ts

```diff
@@ -13,65 +13,81 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import { isObjectType, Element, isInstanceElement } from '@salto-io/adapter-api'
-import { transformValues, TransformFunc } from '@salto-io/adapter-utils'
-import { collections } from '@salto-io/lowerdash'
+import {
+  isObjectType,
+  Element,
+  isInstanceElement,
+  ObjectType,
+} from '@salto-io/adapter-api'
+import { TransformFunc, transformValuesSync } from '@salto-io/adapter-utils'
 import { LocalFilterCreator } from '../filter'
-import { metadataType } from '../transformers/transformer'
-
-const { awu } = collections.asynciterable
+import { apiNameSync, isCustomObjectSync, metadataTypeSync } from './utils'
 
 const TYPE_NAME_TO_FIELD_REMOVALS: Map<string, string[]> = new Map([
   ['Profile', ['tabVisibilities']],
+  [
+    'blng__RevenueRecognitionTreatment__c',
+    ['blng__UniqueId__c', 'blng__Family__c', 'blng__NextOpenPeriod__c'],
+  ],
 ])
 
-const removeFieldsFromTypes = async (
+const fieldRemovalsForType = (
+  type: ObjectType,
+  typeNameToFieldRemovals: Map<string, string[]>,
+): string[] => {
+  const typeName = isCustomObjectSync(type)
+    ? apiNameSync(type) ?? ''
+    : metadataTypeSync(type)
```

Comment:
```suggestion
  const typeName = apiNameSync(type) ?? ''
```
apiName of a type (metadataType / CustomObject) should be correct.